vtex-apps/order-items
# Add new marketingData argument to addToCart function

#### What problem is this solving?

This enables the `addToCart` function exported by `order-items` to receive a `marketingData` argument that is then passed to the `addToCart` mutation.

Without this, the `marketingData` associated with the users current session would not get to the `orderForm` and so it would not be taken into consideration during the checkout process.

The current behavior is seen as a bug and is best described here: https://app.clubhouse.io/vtex/story/32204/utm-source-not-being-passed-to-orderform-when-clicking-the-buy-button

#### How should this be manually tested?

Go to this [Workspace](https://marketingdata--storecomponents.myvtex.com/?utm_source=welcome10), notice that `utm_source` is set to `welcome10`, which triggers a 50% discount for the Gorgeous Top Watch product (should be $220.00). Add this item to the cart and proceed to checkout. You should be charged $220.00 and the discount should be taken into consideration. 

Also, in the Network tab of Chrome DevTools you can see that the `OrderForm` in the `checkout` has the following property:

![Screen Shot 2020-03-04 at 14 59 05](https://user-images.githubusercontent.com/27777263/75908333-b2d18600-5e28-11ea-9279-b7b47245e568.png)

#### Checklist/Reminders


#### Type of changes


✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements 

#### Notes

This depends on:

- https://github.com/vtex/checkout-graphql/pull/48
- (https://github.com/vtex-apps/checkout-resources/pull/32
